src/tools/annotations.R

```diff
@@ -239,8 +239,14 @@ annotate_data <- function(expression_set, chip_id, selections) {
     ))
   }
   
-  # TODO : Maybe print out the number of NAs in the annotations for each
-  # type of annotation?
+  # Print out the number of NAs in the annotations for each type of annotation
+  notMap = matrix(0, nrow = 2, ncol = dim(annotations)[2],
+                  dimnames = list(c("NA entries","%"), colnames(annotations)))
+  for (i in colnames(annotations)) {
+    notMap[1,i] = sum(isNA(annotations[,i]))
+    notMap[2,i] = round(notMap[1,i]/dim(annotations)[1]*1e2, digits = 2)
+  }
+  log_info(paste("Missing annotations:", get.print.ind(notMap), sep = "\n``|| "))
```

Comment:
This looks ok to me! I'd just tweak the indentation, but I'm a nazi on indentations. Additonally, we would need to change the `get.print.ind` function back to `get.print.str` if we scrap the idea, or make it push to the datalog (see my other comment).

Here's the indentation tweak:
```suggestion
  # Print out the number of NAs in the annotations for each type of annotation
  notMap = matrix(
    0, nrow = 2, ncol = dim(annotations)[2],
    dimnames = list(c("NA entries","%"), colnames(annotations))
  )
  for (i in colnames(annotations)) {
    notMap[1,i] = sum(isNA(annotations[,i]))
    notMap[2,i] = round(notMap[1,i]/dim(annotations)[1]*1e2, digits = 2)
  }
  log_info(paste("Missing annotations:", get.print.ind(notMap), sep = "\n``|| "))
```

Reply:
Perfect.
Can we merge to main the first 2 commits and discard the third??
Or should I remove my last pushed commit using e.g. `git revert <commit_hash>` _before_ merging?
What's the rule in these cases?

Reply:
There is a thing called "cherry picking". Look up the git manual, it's pretty easy to use. It lets you "cherry pick" commits to merge.